src/components/Resource/Forms/Training/TrainingAutoTagModal.tsx

```diff
@@ -0,0 +1,185 @@
+import { Button, Group, Input, Stack, Text, Tooltip } from '@mantine/core';
+import JSZip from 'jszip';
+import React, { useState } from 'react';
+import { z } from 'zod';
+import { InfoPopover } from '~/components/InfoPopover/InfoPopover';
+import { createContextModal } from '~/components/Modals/utils/createContextModal';
+import {
+  AutoCaptionType,
+  ImageDataType,
+} from '~/components/Resource/Forms/Training/TrainingImages';
+import { Form, InputNumberSlider, InputSegmentedControl, useForm } from '~/libs/form';
+import { UploadType } from '~/server/common/enums';
+import { useS3UploadStore } from '~/store/s3-upload.store';
+import { showErrorNotification } from '~/utils/notifications';
+import { titleCase } from '~/utils/string-helpers';
+
+const MIN_TAGS = 1;
+export const MAX_TAGS = 30;
+export const MIN_THRESHOLD = 0.3;
+const MAX_THRESHOLD = 0.9;
+
+const OVERWRITE_LIST = ['ignore', 'append', 'overwrite'] as const;
+const OVERWRITES: Record<(typeof OVERWRITE_LIST)[number], string> = {
+  ignore: 'Skip images with existing captions',
+  append: 'Add tags to existing captions',
+  overwrite: 'Overwrite existing captions',
+} as const;
+
+const schema = z.object({
+  maxTags: z.number().int().min(MIN_TAGS).max(MAX_TAGS),
+  threshold: z.number().min(MIN_THRESHOLD).max(MAX_THRESHOLD),
+  overwrite: z.enum(OVERWRITE_LIST),
+});
+export type AutoTagSchemaType = z.infer<typeof schema>;
+
+const defaults = {
+  maxTags: 10,
+  threshold: 0.4,
+  overwrite: 'ignore' as AutoTagSchemaType['overwrite'],
+};
+
+const { openModal, Modal } = createContextModal<{
+  imageList: ImageDataType[];
+  modelId: number;
+  setAutoCaptioning: (modelId: number, data: AutoCaptionType) => void;
+}>({
+  title: 'Automatically caption your images',
+  name: 'autoTag',
+  centered: true,
+  radius: 'lg',
+  Element: ({ context, props: { imageList, modelId, setAutoCaptioning } }) => {
```

Comment:
Briant's implemented a new `dialogStore` that allows us to even have multiple modals open and whatnot. The goal is to migrate away from Mantine's modals, so I'd suggest you use that. 

They work pretty much alike - if you need help with that lmk.